src/render/tutorials/graphql-usage/2_0.html.md

```diff
@@ -0,0 +1,777 @@
+TaskBuilder taskBuilder = new TaskBuilder(FRAGMENT_TYPE, options.getTasks(), proxyProvider);
+```
+
+In order to create a task we need to use `TaskBuilder`. It needs to be supplied with a few things from the configuration. We use `FragmentsHandlerOptions` to parse the configuration. We also need a supplier of action factories. We can retrieve it using standard java `ServiceLoader`. We'll take a look at `supplyFactories` method in a moment.
+
+```java
+Task task = taskBuilder
+        .build(fragment)
+        .orElseThrow(() -> new NullPointerException("No task built"));
+
+return new FragmentEventContextTaskAware(task, eventContext);
+```
+
+Finally we use the task builder to construct a task and return `FragmentEventContextTaskAware` that is passed to the fragments engine in `get` method. Note that in a real application you would probably want to define a custom exception for when task builder fails. If that happens, it's most likely due to wrong or incomplete configuration.
+
+Now let's quickly implement `supplyFactories` method. It simply uses a service loader to retrieve action factories:
+
+```java
+private Supplier<Iterator<ActionFactory>> supplyFactories() {
+    return () -> {
+        ServiceLoader<ActionFactory> factories = ServiceLoader.load(ActionFactory.class);
+        return factories.iterator();
+    };
+}
+```
+
+#### Retrieving payload
+Now that we can make our fetcher execute Knot.x tasks from configuration, we need to retrieve the outcome. As noted before, the data retrieved from 3rd party (google books api in our case) is stored in the outcome fragment's payload. Therefore our final `get` method should look like this:
+
+```java
+@Override
+public CompletableFuture<T> get(DataFetchingEnvironment environment) {
+    FragmentEventContextTaskAware eventContextTaskAware = setupTask(vertx, config, routingContext, environment);
+    CompletableFuture<T> future = new CompletableFuture<>();
+
+    engine
+        .execute(Collections.singletonList(eventContextTaskAware))
+        .subscribe(events -> {
+            JsonObject payload = events.get(0).getFragment().getPayload();
+            JsonObject fetchedData = payload.getJsonObject("fetchedData");
+            T model = getDataObjectFromJson(fetchedData, environment);
+            future.complete(model);
+        });
+
+    return future;
+}
+```
+
+After the engine is done executing the task tree we:
+- retrieve the fragment and its payload
+- get data from payload under "fetchedData" key
+- transform this data into our model (in our case it would be `Book`)
+- complete the future with our model so that GraphQL gets the data
+
+There are two things to explain here. Firstly, we assume that the data we need is under "fetchedData" key. It is not exacly the case but we'll deal with it later. Secondly, we use `getDataObjectFromJson` method to transform json into our model. This method is abstract because our `TaskDataFetcher` don't know how to do it.

```

Comment:
There are few repetitions of sentences like:
> There are things to explain here...

Maybe it would be worth to add some alternatives like:
> So what just happened? We assumed...
> Some explanation to the code above...
> What we did above was...